src/Servers/IIS/IIS/test/Common.FunctionalTests/OutOfProcess/AspNetCorePortTests.cs

```diff
@@ -84,6 +84,48 @@ public async Task EnvVarInWebConfig_Invalid(TestVariant variant, string port)
             Assert.Equal(HttpStatusCode.BadGateway, response.StatusCode);
         }
 
+        [ConditionalTheory]
+        [MemberData(nameof(TestVariants))]
+        [RequiresNewShim]
+        public async Task ShutdownMultipleTimesWorks(TestVariant variant)
+        {
+            // Must publish to set env vars in web.config
+            var deploymentParameters = Fixture.GetBaseDeploymentParameters(variant);    
+
+            var deploymentResult = await DeployAsync(deploymentParameters);
+            
+            // Shutdown once
+            var response = await deploymentResult.HttpClient.GetAsync("/Shutdown");
+            
+            // Wait for server to start again.
+            for (var i = 0; i < 10; i++)
+            {
+                // ANCM should eventually recover from being shutdown multiple times.
+                response = await deploymentResult.HttpClient.GetAsync("/HelloWorld");
+                if (response.IsSuccessStatusCode)
+                {
+                    break;
+                }
+            }
+            
+            // Shutdown again
+            response = await deploymentResult.HttpClient.GetAsync("/Shutdown");
```

Comment:
Was it the first shutdown that causes the second shutdown to never finish?

Reply:
Yep.

Reply:
Well, it's more the first shutdown hangs and the second shutdown never starts because of it.